app/views/posts/index.html.erb

```diff
@@ -0,0 +1,24 @@
+<h1>All Posts</h1>
+
+<ul>
+  <% @posts.each do |post| %>
+    <li>
+      <%= link_to post.title, posts_show_path(post), class: "post_title" %>
+      <p class = "pl-2">
+      <i>by: 
+      <%  if logged_in? %> 
+        <%= User.find(post.user_id).name %>
```

Comment:
You shouldn't use application logic in any of the views. Please remove here and in any other view.